src/Veldrid/GraphicsPipelineDescription.cs

```diff
@@ -160,7 +160,9 @@ public bool Equals(GraphicsPipelineDescription other)
                 && PrimitiveTopology == other.PrimitiveTopology
                 && ShaderSet.Equals(other.ShaderSet)
                 && Util.ArrayEquals(ResourceLayouts, other.ResourceLayouts)
-                && ResourceBindingModel.Equals(other.ResourceBindingModel)
+                && (ResourceBindingModel.HasValue && other.ResourceBindingModel.HasValue
```

Comment:
Should this use Util.NullableEquals?

Reply:
It's an enum, and Util.NullableEquals has an IEquatable<T> constraint. We could remove the constraint and use EqualityComparer<T>.Default to compare the values, but that would make it as slow as Nullable.Equals<T> (since that's exactly how it's implemented).

|         Method |      Mean |     Error |    StdDev |
|--------------- |----------:|----------:|----------:|
| CustomEquality |  2.430 ns | 0.0131 ns | 0.0122 ns |
| NullableEquals | 14.788 ns | 0.1120 ns | 0.1048 ns |

It's convenient to use pipeline state as a dictionary key (even though there's probably options that involve less hashing and comparsions), so I think Equals should be as fast as possible.

Reply:
Okay, that makes sense. I'd like to update to C# 7.3 at some point, but it isn't really worth it for just this one change. This is good as-is.